pkg/encoding/fix_test.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encoding
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/kedgeproject/kedge/pkg/spec"
+
+	api_v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func TestFixServices(t *testing.T) {
+	failingTest := []spec.ServiceSpecMod{
```

Comment:
>How does it become difficult to understand, there is a field right there next to the test which says if the test is expected to pass or fail.

The part that becomes difficult to understand is the code that runs test for example: 

```go
	for _, test := range tests {
		t.Run(test.Name, func(t *testing.T) {
			kd, err := transformer.CreateDeployments(test.Service)
			if err != nil {
				if test.Succeed {
					t.Errorf("Failed to create deployment from %#v\nErr: %s", test.Service, err)
				}
				return
			}


			if !test.Succeed {
				t.Errorf("Expected failure, but succeeded, service: %#v\nConverted k8s Deployment: %#v", test.Service, spew.Sprint(kd))
				return
			}


			if !reflect.DeepEqual(kd, test.K8sDeployments) {
				t.Errorf("Expected: %#v\nGot: %#v\n", spew.Sprint(test.K8sDeployments), spew.Sprint(kd))
				return
			}
		})
	}
```


taken from opencompose in [here](https://github.com/redhat-developer/opencompose/blob/a2718d7320c28eb379830235cb2f67b13351ed14/pkg/transform/kubernetes/kubernetes_test.go#L677-L694)

If you are checking condition on `test.Succeed` is used twice which goes out of the window when we have separated the two things.


And see the code we have here:

```go
	for _, test := range passingTests {
		t.Logf("Running test: %s", test.Name)
		got, err := fixServices(test.Input, appName)
		if err != nil {
			t.Errorf("expected to pass but failed with: %v", err)
		}
		if !reflect.DeepEqual(got, test.Output) {
			t.Errorf("expected: %s, got: %s", prettyPrintObjects(test.Output),
				prettyPrintObjects(got))
		}
	}
```
source [here](https://github.com/surajssd/kedge/blob/fe8ed8e1b5f11d9cd26f2eb5ca03b9f43892423f/pkg/encoding/fix_test.go#L82-L92)

No convoluted logic of `test.Succeed` here. Because we know test should pass and all output should match as well.


And for failing tests we just need to see if it failed:

```go
	got, err := fixVolumeClaims(passingTest, appName)
	if err != nil {
		t.Errorf("expected to pass but failed with: %v", err)
	}
	if !reflect.DeepEqual(got, expected) {
		t.Errorf("expected: %s, got: %s", prettyPrintObjects(expected),
			prettyPrintObjects(got))
	}
```

source [here](https://github.com/surajssd/kedge/blob/fe8ed8e1b5f11d9cd26f2eb5ca03b9f43892423f/pkg/encoding/fix_test.go#L108-L115)